language-design.md

```diff
@@ -28,28 +102,114 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
-
+It is much easier to spot a poorly-designed language than a well-designed
+language. While there are certain principles that can help you spot a
+well-designed language, time may be a factor in discovering a languages
+shortcomings.
+
+A well designed language should be approchable and "self-documenting." Just like
+APIs you should be able to understand the code without constantly looking at
+documentation once you know the pattern. This leads me to a second point:
+well-designed languages must have an overall pattern as the first paper
+mentioned [Steele, 1998]. Furthermore, as Steele explained, language extenstions
+should feel natural and fit within the rest of the language. This provides a way
+of future-proofing a language. It also serves to make the language more readable
+and writable since patterns can be better maintained.
+
+Bad languages are either hard to read or hard to write. For example, in Verou's
+article regarding the choice of naming a "grayness" function, they raised the
+issue of calling such a function `gray` with a percentage value since it is
+not obviuous which means white and which means black. A lack of intuitiveness
+in language makes it bad; intuitiveness coupled with expected behavior can make
+it better.
+
+Efficiency with features is key to a well-designed language. It's not helpful
```

Comment:
This also seemed to be the point made in Tim Smith's guide to R, when he mentions that there are too many ways to assign a variable in R, and this can be quite confusing to the user. But then others would argue that this kind of flexibility can be helpful for users. Again, I suppose there always need to be some sort of tradeoff, between flexibility/efficiency, readability/writeability, etc.